datafusion/core/tests/sql/window.rs

```diff
@@ -471,3 +471,297 @@ async fn window_with_agg_in_expression() -> Result<()> {
+//     let sql = "SELECT \
+//                SUM(c4) OVER(PARTITION BY c1 ROWS BETWEEN 1 PRECEDING AND 1 FOLLOWING),\
+//                COUNT(*) OVER(PARTITION BY c2 ROWS BETWEEN 1 PRECEDING AND 1 FOLLOWING)\
+//                FROM aggregate_test_100 \
+//                ORDER BY c9 \
+//                LIMIT 5";
+//     let actual = execute_to_batches(&ctx, sql).await;
+//     let expected = vec![
+//         "+----------------------------+-----------------+",
+//         "| SUM(aggregate_test_100.c4) | COUNT(UInt8(1)) |",
+//         "+----------------------------+-----------------+",
+//         "| -33822                     | 3               |",
+//         "| 20808                      | 3               |",
+//         "| -29881                     | 3               |",
+//         "| -47613                     | 3               |",
+//         "| -13474                     | 3               |",
+//         "+----------------------------+-----------------+",
+//     ];
+//     assert_batches_eq!(expected, &actual);
+//     Ok(())
+// }
+
+#[tokio::test]
+async fn window_frame_rows_preceding_with_unique_partition() -> Result<()> {
+    let ctx = SessionContext::new();
+    register_aggregate_csv(&ctx).await?;
+    let sql = "SELECT \
+               SUM(c4) OVER(PARTITION BY c9 ROWS BETWEEN 1 PRECEDING AND 1 FOLLOWING),\
+               COUNT(*) OVER(PARTITION BY c9 ROWS BETWEEN 1 PRECEDING AND 1 FOLLOWING)\
+               FROM aggregate_test_100 \
+               ORDER BY c9 \
+               LIMIT 5";
+    let actual = execute_to_batches(&ctx, sql).await;
+    let expected = vec![
+        "+----------------------------+-----------------+",
+        "| SUM(aggregate_test_100.c4) | COUNT(UInt8(1)) |",
+        "+----------------------------+-----------------+",
+        "| -16110                     | 1               |",
+        "| 3917                       | 1               |",
+        "| -16974                     | 1               |",
+        "| -1114                      | 1               |",
+        "| 15673                      | 1               |",
+        "+----------------------------+-----------------+",
+    ];
+    assert_batches_eq!(expected, &actual);
+    Ok(())
+}
+
+#[tokio::test]
+async fn window_frame_ranges_preceding_following() -> Result<()> {
+    let ctx = SessionContext::new();
+    register_aggregate_csv(&ctx).await?;
+    let sql = "SELECT \
+               SUM(c4) OVER(ORDER BY c2 RANGE BETWEEN 1 PRECEDING AND 1 FOLLOWING),\
+               SUM(c3) OVER(ORDER BY c2 RANGE BETWEEN 10000 PRECEDING AND 10000 FOLLOWING),\
+               COUNT(*) OVER(ORDER BY c2 RANGE BETWEEN 1 PRECEDING AND 1 FOLLOWING) \
+               FROM aggregate_test_100 \
+               ORDER BY c9 \
+               LIMIT 5";
+    let actual = execute_to_batches(&ctx, sql).await;
+    let expected = vec![
+        "+----------------------------+----------------------------+-----------------+",
```

Comment:
I wonder if you ran the same queries in postgres? If not, no worries, but I am having a hard time validating these answers are correct (mostly because the `register_aggregate_csv` isn't a great dataset lol)